### PR TITLE
fix(MAINTAINERS): add a maintainers file

### PR DESCRIPTION
**Pull request by technosophos**
This supercedes #279.

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+* Ville Aikas - [@vaikas-google](https://github.com/vaikas-google)
+* Matt Butcher - [@technosophos](https://github.com/technosophos)
+* Jack Greenfield - [@jackgr](https://github.com/jackgr)
+* Brendan Melville - [@bmelville](https://github.com/bmelville)
+* Michelle Noorali - [@michelleN](https://github.com/michelleN)
+* Adam Reese - [@adamreese](https://github.com/adamreese)
+
+### Code Reviews
+
+Before being merged, every pull request must have at least two LGTMs ("Looks Good To Me"), and at least one of these must come from one of the core maintainers listed above.
```
Comment (technosophos) [inline MAINTAINERS.md]: There are two alternative proposals on the table for this one:

- Drop to one LGTM (which must come from a core maintainer)
- Follow the Kubernetes guidelines:
  - One LGTM from a core maintainer for small commits (~10 lines?)
  - Two LGTMs for other commits, one of which must come from a core contributor

File: MAINTAINERS.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+## Becoming a Core Maintainer
+
+Generally, potential core maintainers are selected by the Helm core maintainers based in
+part on the following criteria:
+
+* Sustained contributions to the project over a period of time (usually months)
+* A willingness to help users on GitHub and in the Helm Slack room for [Kubernetes](http://slack.kubernetes.io/)
+* A friendly attitude
+
+The Helm core maintainers must unanimously agree before inviting a community member to join as a maintainer, although in many cases the candidate has already been acting in the capacity of a contributing maintainer for some time, and has been consulted on issues, pull requests, etc.
```
Comment (technosophos) [inline MAINTAINERS.md]: There are two open questions on this one:

- Is there a process better than unanimous vote?
- Should we add a second "class" of maintainer that is not a core maintainer, but is still somehow more of a contributor than a typical community member?

File: MAINTAINERS.md (hunk 1 above)
Comment (michelleN) [inline MAINTAINERS.md]: My thoughts:

1. LGTM:
pro: fast merges 
con: less knowledge transfer about larger PRs that might have to with making an architectural decision

2. The Kubernetes pattern 
pro: allows for greater knowledge transfer within the group
con: might be overkill have to find 2 people to LGTM before merging might take time unless we use something like mention-bot (https://github.com/facebook/mention-bot)

For helm and at deis, we do 2 LGTMs. The Kubernetes pattern is also 2 LGTMs for large PRs. I think it'd be nice to do at least 2 LGTMs on large PRs, but if the rest of the team doesn't care for it, I'm okay with 1.

File: MAINTAINERS.md (hunk 2 above)
Comment (michelleN) [inline MAINTAINERS.md]: It makes sense to have a `contributing maintainer` category for heavily contributing members of the community. I think we all would value regular contributions and feedback from folks who are using helm consistently. It also makes sense to me to have consensus from the team when adding these maintainers if we go that route. We could also just cross that bridge when we get to it if we wanted.

Comment (michelleN) [inline MAINTAINERS.md]: cc/ @sparkprime @bmelville @vaikas-google @adamreese @jackgr any thoughts?

File: MAINTAINERS.md (hunk 1 above)
Comment (michelleN) [inline MAINTAINERS.md]: cc/ @sparkprime @bmelville @vaikas-google @adamreese @jackgr any thoughts?

Comment (jackgr) [inline MAINTAINERS.md]: So, we've been using 1 LGTM due to the small size of our team. Also, for minor changes, it seems like overkill to require 2 LGTMs. Keep in mind that Kubernetes has about 20 active committers.

In the trade-off between big up front design and rapid iteration, I'm always a fan of the latter. For that reason, keeping the velocity up matters more to me than being 100% certain of what we're committing. Experience suggests that 100% certainty is a myth, anyway.

Worst case, if a committer pushes something that we really don't want, and the 1 reviewer doesn't catch it, we can always back it out. That has happened many times in k8s land, despite their 2 LGTMs. It's a complex product with many opportunities for side effects. Ours is simpler, so I don't think we'll see that need to back out commits as often as they do.

File: MAINTAINERS.md (hunk 2 above)
Comment (jackgr) [inline MAINTAINERS.md]: In any organization, ASF being the canonical example, there's always a core team that "owns" the project. That's also true in Kubernetes. So, not all committers/maintainers are created equal.

That said, I don't see the need to formalize it. Less is generally more when it comes to process, IMO. Besides, only a subset of the maintainers have team admin rights, anyway. So, in practice, it doesn't really matter what we call people. It matters whether or not they have admin rights.

As for unanimous vote, it's just the sound of it that bothers me. If this were a large and bureaucratic organization, then perhaps that would be necessary, but let's be honest with ourselves here... 

If one of us with admin rights were to admit someone who didn't belong, the rest of us would let him/her know about it. :0) Likewise, if one of us without admin rights feels strongly that we should add someone to the committers team, then those of us with admin rights would be obliged to seriously consider their request and to grant it barring any legitimate concerns.

File: MAINTAINERS.md (hunk 1 above)
Comment (michelleN) [inline MAINTAINERS.md]: I'll be fine with 1 lgtm.

File: MAINTAINERS.md (hunk 2 above)
Comment (michelleN) [inline MAINTAINERS.md]: I'd be fine with continuing in this manner. 
However, I will respectfully disagree with the idea that if this were a large and bureaucratic organization that we would need a unanimous vote. I think those are more realistic with small teams like ours. I just like the idea of consensus but I do think there is enough trust here to be fine with letting admins add and remove core team members as they and the team see fit.

Comment (jackgr) [inline MAINTAINERS.md]: Not sure, but I think we're in agreement... No unanimous vote is needed, yes?

Comment (michelleN) [inline MAINTAINERS.md]: yes.

Comment (jackgr) [inline MAINTAINERS.md]: yay.

File: Makefile [hunk 3]
```diff
@@ -73,12 +73,26 @@ vet:
 	done
 	@echo -----------------
 
+HAS_GLIDE := $(shell command -v glide)
```
Comment (jackgr) [inline Makefile]: Why are these makefile changes in the PR for the maintainers document???

Comment (technosophos) [inline Makefile]: That is a good question. Those were in a separate PR already merged onto master. I'll see if a rebase on this branch fixes it.

**Comment by technosophos** [comment]
I'll close, cherry-pick, and open a new PR.

**Comment by jackgr** [comment]
Perfect.

**Pull request closed without merging**